ts/store/reducers/pinlogin.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * A reducer for the PinLogin.
+ * @flow
+ */
+
+import { PIN_LOGIN_VALIDATE_FAILURE } from "../actions/constants";
+import { Action } from "../actions/types";
+
+export type PinLoginState = Readonly<{
+  PinConfirmed: string;
```

Comment:
instead of `string`, the type of `PinConfirmed` should be `"PinCheck" | "PinConfirmedInvalid"`